util/deploy.sh

```diff
@@ -23,30 +27,67 @@ info "Installing dependencies..."
 cd ${WPTD_PATH}; make go_deps;
 
 # Create a name for this version
-BRANCH_NAME="$(git rev-parse --abbrev-ref HEAD)"
-USER="$(git remote -v get-url origin | sed -E 's#(https?:\/\/|git@)github.com(\/|:)##' | sed 's#/.*$##')"
-VERSION="${USER}-${BRANCH_NAME}"
+BRANCH_NAME=${TRAVIS_PULL_REQUEST_BRANCH:-"$(git rev-parse --abbrev-ref HEAD)"}
+USER="$(git remote -v get-url origin | sed -E 's#(https?:\/\/|git@)github.com(\/|:)##' | sed 's#/.*$##')-"
+if [[ "${USER}" == "web-platform-tests" ]]; then USER=""; fi
+
+VERSION="${USER}${BRANCH_NAME}"
 PROMOTE="--no-promote"
 
 if [[ ${PRODUCTION} == 'true' ]]
 then
   info "Producing production configuration..."
-  if [[ "${USER}" != "w3c" ]]
+  if [[ "${USER}" != "web-platform-tests" ]]
   then
-    confirm "Are you sure you want to be deploying a non-w3c repo (${USER})?"
-    if [ "${?}" != "0" ]; then exit "${?}"; fi
+    if [[ "${QUIET}" != "true" ]]
+    then
+      confirm "Are you sure you want to be deploying a non-web-platform-tests repo (${USER})?"
+      if [ "${?}" != "0" ]; then exit "${?}"; fi
+    fi
   fi
   # Use SHA for prod-pushes.
   VERSION="$(git rev-parse --short HEAD)"
   PROMOTE="--promote"
 fi
 
-COMMAND="gcloud app deploy ${PROMOTE} --version=${VERSION} ${WPTD_PATH}/webapp"
+if [[ "${QUIET}" == "true" ]]
+then
+    QUIET_FLAG="-q"
+else
+    QUIET_FLAG=""
+fi
+COMMAND="gcloud app deploy ${PROMOTE} ${QUIET_FLAG} --version=${VERSION} ${WPTD_PATH}/webapp"
 
 info "Deploy command:\n${COMMAND}"
-confirm "Execute?"
-if [ "${?}" != "0" ]; then exit "${?}"; fi
+if [[ "${QUIET}" != "true" ]]
+then
+    confirm "Execute?"
+    if [ "${?}" != "0" ]; then exit "${?}"; fi
+fi
 
 info "Executing..."
-${COMMAND}
-exit 0
+${COMMAND} || exit $?
```

Comment:
If you find yourself repeating this pattern, you can try to add `set -e` at the top of your Bash script, which will exit the script immediately if any command fails. Then, for the commands that you think may (and can) fail, add `|| true` at the end of those lines instead.

(In fact, I'd recommend doing so, but I'm not sure if it'll break anything existing; you can try it out.)